main.go

```diff
@@ -0,0 +1,96 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+
+	"github.com/dispatchframework/funky/pkg/funky"
+)
+
+const (
+	serversEnvVar   = "SERVERS"
+	serverCmdEnvVar = "SERVER_CMD"
+)
+
+type funkyHandler struct {
+	router funky.Router
+}
+
+func (f funkyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	reader := r.Body
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(reader)
+	rawBody := buf.Bytes()
+	var body map[string]interface{}
```

Comment:
Should be of `*Message` type (introduced in an earlier comment). As a free benefit, message structure will be enforced. Also, we're going to use some context fields, so it's gonna be more convenient.